to.etc.domui/src/main/java/to/etc/domui/component/layout/TabPanelBase.java

```diff
@@ -282,8 +284,15 @@ private TabInstance addTabInstance(TabInstance ti, int pos) {
 			m_tablist.add(pos, ti);
```

Comment:
Something is wrong here in the existing code.
If the "pos" is out of bounds the thing gets added as the last node. But "pos" remains "wrong". The code should become something like this I think:

    if(pos < 0 || pos >= m_tabList.size()) {
      pos = m_tabList.size();
    }
    m_tabList.add(pos, ti);

In addition, the m_currentTab field is not correctly updated: if we insert at or before m_currentTab then m_currentTab must be incremented. I.e. if currentTab is 3, and we insert at 2, then after the insert currentTab is 4.

So

    if(pos <= m_currentTab)
      m_currentTab++;